Riot/Modules/Settings/SettingsViewController.m

```diff
@@ -340,6 +340,18 @@ - (void)finalizeInit
     self.screenTracker = [[AnalyticsScreenTracker alloc] initWithScreen:AnalyticsScreenSettings];
 }
 
+- (void)dealloc {
+    // Remove observers
+    if (pushInfoUpdateObserver)
+    {
+        [[NSNotificationCenter defaultCenter] removeObserver:pushInfoUpdateObserver name:kMXKAccountAPNSActivityDidChangeNotification object:nil];
+        pushInfoUpdateObserver = nil;
+    }
+
+    // Fix for destroy not being called
+    [self destroy];
```

Comment:
```suggestion
    // Fix for destroy not being called
    [self destroy];
```

At this point, why we don't call just `destroy`?
Inside `destroy` there is a call to `reset` which should remove the observer.

Reply:
Yes, we can as `pushInfoUpdateObserver` is removed in destroy.

Done.